File: backend/src/main/java/com/smvita/computerseekho/dto/ApiError.java
package com.smvita.computerseekho.dto;

import java.time.Instant;
import java.util.List;

/**
 * Uniform error shape for every non-2xx response. The frontend's
 * axiosClient interceptor reads `message` specifically — keep that field
 * name stable so every screen's error handling keeps working.
 */
public record ApiError(
        Instant timestamp,
        int status,
        String message,
        List<String> fieldErrors
) {
    public ApiError(int status, String message) {
        this(Instant.now(), status, message, null);
    }

    public ApiError(int status, String message, List<String> fieldErrors) {
        this(Instant.now(), status, message, fieldErrors);
    }
}

File: dotnet_backend/backend/DTO/ApiError.cs
namespace ComputerSeekho.DTO;

/// <summary>
/// The single error shape every failed request returns.
///
/// Matches the Java backend's ApiError field for field, because the shared
/// React client already parses it: axiosClient reads `message` and joins
/// `fieldErrors` onto it. Returning ASP.NET Core's default
/// ProblemDetails instead would leave every error on the frontend reading
/// "Something went wrong", since the property names differ.
/// </summary>
public record ApiError(
    int Status,
    string Message,
    IEnumerable<string>? FieldErrors = null
);
